astropy/table/sorted_array.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+
+class SortedArray(object):
+    '''
+    Implements a sorted array container using
+    a list of numpy arrays.
+
+    Parameters
+    ----------
+    lines : `Table`
```

Comment:
> Currently data is a list of ndarrays rather than Column objects. I could definitely turn it into a Table if that makes the code easier to read/maintain, although my initial thought was that an ndarray will be faster, e.g. since Column has a custom __setitem__.

This strategy doesn't play well with mixin columns because you end up with an ndarray of objects, which will blow up for long tables:
```
t = simple_table(10)
t['t'] = Time(t['b'], format='cxcsec')
type(t['t'])
t.add_index('t')
t.indices[0]
[[<Time object: scale='tt' format='cxcsec' value=1.0>
 <Time object: scale='tt' format='cxcsec' value=2.0>
 <Time object: scale='tt' format='cxcsec' value=3.0>
 <Time object: scale='tt' format='cxcsec' value=4.0>
 <Time object: scale='tt' format='cxcsec' value=5.0>
 <Time object: scale='tt' format='cxcsec' value=6.0>
 <Time object: scale='tt' format='cxcsec' value=7.0>
 <Time object: scale='tt' format='cxcsec' value=8.0>
 <Time object: scale='tt' format='cxcsec' value=9.0>
 <Time object: scale='tt' format='cxcsec' value=10.0>], [0 1 2 3 4 5 6 7 8 9]]
```